docs/split-inputs.md

```diff
@@ -119,9 +119,70 @@ It is relatively easy for a modern fuzzing engine to discover the separator,
 but nevertheless we recommend to provide several seed inputs with the desired number
 of separators.
 
-## Fuzz Data Provider
-
-TODO
+## Fuzzed Data Provider
+
+[FuzzedDataProvider] is a single-header C++ library that is helpful for
+splitting a fuzz input into multiple parts of various types. Both advantage and
+disadvantage of using this library is that the input splitting happens
+dynamically, i.e. you don't need to define any structure of the input. This
+might be very helpful in certain cases, but could also make the corpus human
+unreadable.
+
+[FuzzedDataProvider] is a class whose constructor accepts `const uint8_t,
+size_t` arguments. Below is a quick overview of the available methods.
+
+### Methods for extracting individual values
+
+* `ConsumeBool`, `ConsumeIntegral`, `ConsumeIntegralInRange` methods are helpful
+  for extracting a single boolean or integer value, e.g. some flag for the
```

Comment:
Would it make sense to specify here the sizes of the integers? Or this is something documented in more detail in the API comments?

Reply:
I didn't want to get into too much detail, but it feels like it would be better to mention that return value type is a template parameter. Otherwise, readers might think that it always returns an `int`. Done.